### ISSUE TITLE
Crash influence on the part's age

### ISSUE THREAD

**Issue by Ippo343** [Ippo343/DangIt#20]
A collision should cause the part to age proportionally to the violence of the impact.

Specifically, I mean to increase the part's age by multiplying it by the factor:

```
 f = 1 + (c * v / V)
```

where v is the impact velocity, V the crash tolerance, and c a sensitivity factor.

**Comment by 602p** [comment]
We can add a `OnCollisionEnter` hook to the `FailureModule` class and then find `collision.relativeVelocity` to compute the effect on age.

Perhaps we then store the relative velocity and return it in a `CrashModifier` function:

```mono
void OnCollisionEnter(Collision collision) {
    this.lastCollisionVelocity = collision.relativeVelocity;
}
```

```mono
private float CrashMultiplier(){
    float value = 1 + Collision.relativeVelocity / crashTolerance;
    this.lastCollisionVelocity = 0;
    return value;
}
```

and then replace [FailureModule.cs#L427](https://github.com/Ippo343/DangIt/blob/master/source/DangIt/Failure%20modules/FailureModule.cs#L427) with

```mono
this.Age += (dt * this.TemperatureMultiplier() * this.CrashMultiplier());
```

What do you think?

**Comment by Ippo343** [comment]
Well, a collision is a discrete event, not something that happens continuously (hopefully :P ), so there's no need to handle it in Update(): we can simply just bump the age once in `onCollisionStay`

    void OnCollisionStay(Collision collision) {
        this.Age += some_function_of(collision.relativeVelocity);
    }

Also, judging only from the names, are you sure we don't want this to be in onCollisionEnter?

Sorry for being so absent, I'm working ~60 hours per week :/

**Comment by 602p** [comment]
Good point. The reason I was thinking it would go in `onCollisonStay` instead of `onCollisionEnter` is so that you still get age debuffs for scraping along the ground. This of course would need some balance testing...

On the note of what would be in `some_function_of` I think we could use

`this.Age += this.LifetimeSecs * ( (collision.relativeVelocity / this.part.maxImpactSpeed) / IMPACT_CONSTANT )`

I think a good value for `IMPACT_CONSTANT` would be ~2? That way a collision at 1/10th of the max speed of the part would add 5% to its age.
